infra/modules/aws-ssm-secrets/main.tf

```diff
@@ -41,17 +38,17 @@ resource "aws_ssm_parameter" "secret" {
 resource "aws_ssm_parameter" "secret_with_externally_managed_value" {
   for_each = local.externally_managed_secrets
 
-  name = "${local.path_prefix}${each.key}"
-  # Due https://github.com/hashicorp/terraform-provider-aws/issues/31267
-  # all are added as secureString
-  type        = "SecureString"
-  description = each.value.description
-  value       = sensitive(coalesce(each.value.value, local.PLACEHOLDER_VALUE))
-  key_id      = coalesce(var.kms_key_id, "alias/aws/ssm")
+  name           = "${local.path_prefix}${each.key}"
+  type           = each.value.sensitive ? "SecureString" : "String"
+  description    = each.value.description
+  value          = each.value.sensitive ? sensitive(coalesce(each.value.value, local.PLACEHOLDER_VALUE)) : null
+  insecure_value = each.value.sensitive ? null : coalesce(each.value.value, local.PLACEHOLDER_VALUE)
+  key_id         = coalesce(var.kms_key_id, "alias/aws/ssm")
 
   lifecycle {
     ignore_changes = [
       value, # key difference here; we don't want to overwrite values filled by the external process
+      insecure_value,
```

Comment:
What happens if for example `client_id` is wrongly provided if is not sensitive? Does `value` will update `insecure_value` on second attempt ?

Reply:
if this is in lifecycle, then TF shouldn't change resource even if it believes changes are needed in EITHER `value` or `insecure_value`